### ISSUE TITLE
Redirect coverage.nodejs.org to codecov.io

### ISSUE THREAD

**Issue by bcoe** [nodejs/nodejs.org#3583]
Are we happy enough with [codecove.io's coverage tracking](https://codecov.io/gh/nodejs/node) that we could consider redirecting `coverage.nodejs.org` to its results?

The advantage being that our nightly coverage jobs have had a habit of breaking throughout the years.

CC: @mhdawson, @Trott

**Comment by mhdawson** [comment]
I'll defer to @Trott on this one since he probably looks at the coverage output on a more regular basis. If he's comfortable then I'd be ok as well.

Having said that I'm trying to look at some of the results and getting this:

![image](https://user-images.githubusercontent.com/9373002/102655172-d8bde200-413f-11eb-858b-b88484a0962f.png)

**Comment by bcoe** [comment]
CC: @thomasrockhu we still seem to be having some oddities with displaying reports occasionally:

Seems to work for me:

<img width="1792" alt="Screen Shot 2020-12-18 at 12 15 12 PM" src="https://user-images.githubusercontent.com/194609/102657550-b8cff380-412a-11eb-8f03-90e53308bc82.png">

**Comment by mhdawson** [comment]
Yes, I looked at one ok, but then when I selected another I got the error I showed. After that I could not find one that would display.

**Comment by slimcoders** [comment]
Guy's what this coverage is all about? Code which we wrote and use right? Or it is related to testing.
Please tell I'm interested to work on this.

**Comment by thomasrockhu** [comment]
@mhdawson, @bcoe looks fine on my end as well (logged in and out). If you come across another of if this is still an issue, let me know!

**Comment by mhdawson** [comment]
I still get this from 2 different machines. I even tried using firefox and still get the same thing:

https://codecov.io/gh/nodejs/node/src/master/lib/_http_incoming.js

![image](https://user-images.githubusercontent.com/9373002/102795059-40f80800-437a-11eb-9f82-a24fd1872de0.png)

**Comment by mhdawson** [comment]
@thomasrockhu FYI, seems to be an ongoing issue, not sure why its affects some but not all.

**Comment by thomasrockhu** [comment]
Thanks @mhdawson, looks like there is a bug on our end. Our current suspicion is that users that are not logged in cannot access the page. This is NOT the intended behavior, and we will work on a fix.

Please note that it is the holiday season, so it may take some time to roll out the patch as a significant portion of our team is off.

**Comment by bcoe** [comment]
> Guy's what this coverage is all about? Code which we wrote and use right? Or it is related to testing.
Please tell I'm interested to work on this.

Hey @slimcoders, we track the coverage of the lines of code that unit tests in the Node.js codebase touch. These reports can represent a good starting point for people to make contributions to Node.js, by writing unit tests for parts of the codebase that are missing coverage, see:

https://coverage.nodejs.org/

For reports.

**Comment by slimcoders** [comment]
I observe that there are so many lines not covered. so i can work on that?

**Comment by mhdawson** [comment]
@slimcoders that is one of the reasons we generate/publish the coverage info. Many PRs to increase coverage have come from that so it is a good way to contribute :)

**Comment by Trott** [comment]
One way to a quick resolution might be a landing page that links to both. It will be an extra click, but I think that would be OK. If we could do anonymized click-tracking on the page to see which link people use more often, even better.

I'd want the following questions answered if we were to switch directly to codecov.io at this time:

* How would people get to the existing coverage.nodejs.org? Would it be completely inaccessible?
* The C++ coverage difference between the two is very large. What's the explanation for that? 
* In the existing coverage, I almost exclusively look at and sort by branch coverage instead of line coverage. I can try to switch my workflow, but I'd rather not if I don't have to. I think branch coverage is the most useful metric. Any chance at all that will be implemented in codecov.io? Is it an option I need to enable or something?

**Comment by Trott** [comment]
I just tried looking at codecov.io stats in detail for the first time in a while and it looks like maybe our config needs a tweak? Clicking through to JavaScript files doesn't show the source but shows an error message instead. When I end up at https://codecov.io/gh/nodejs/node/src/master/lib/async_hooks.js, I get a GitHub API Forbidden error:

![image](https://user-images.githubusercontent.com/718899/103101618-5618a500-45cd-11eb-825a-40ef5cd1d111.png)

**Comment by richardlau** [comment]
@Trott see https://github.com/nodejs/nodejs.org/issues/3583#issuecomment-749137557.

**Comment by Trott** [comment]
Seems like my branches question can be ignored. The way the line coverage vs. partial line coverage stuff is displayed in codecov.io seems to surface the same information in a different way. I can use that.

**Comment by XhmikosR** [comment]
IIRC the issue should be part of the build repo since that's where the redirects happen.
